core/frontend/src/quantity-formatting/QuantityFormatter.ts

```diff
@@ -586,6 +608,40 @@ export class QuantityFormatter implements UnitsProvider {
   public async onInitialized() {
     await this.initializeQuantityTypesRegistry();
 
+    const initialKoQs = [["AecUnits.LENGTH", "Units.M"], ["AecUnits.ANGLE", "Units.RAD"], ["AecUnits.AREA", "Units.SQ_M"], ["AecUnits.VOLUME", "Units.CUB_M"]];
+    for (const entry of initialKoQs) {
+      try {
+        await this.addFormattingSpecsToRegistry(entry[0], entry[1]);
+      } catch (err) {
+        console.log(err);
+      }
+    }
+    this._removeFormatsProviderListener = IModelApp.formatsProvider.onFormatsChanged.addListener(async (args: FormatsChangedArgs) => {
+      if (args.formatsChanged === "all") {
+        this._formatSpecsRegistry.forEach(async (entry, name) => {
+          const formatProps = await IModelApp.formatsProvider.getFormat(name);
+          if (formatProps) {
+            const persistenceUnitName = entry.formatterSpec.persistenceUnit.name;
+            await this.addFormattingSpecsToRegistry(name, persistenceUnitName, formatProps);
+          } else {
+            this._formatSpecsRegistry.delete(name); // clear the specs if format is removed
+          }
+        });
+      } else {
+        for (const name of args.formatsChanged) {
+          if (this._formatSpecsRegistry.has(name)) {
+            const formatProps = await IModelApp.formatsProvider.getFormat(name);
+            if (formatProps) {
+              const persistenceUnitName = this._formatSpecsRegistry.get(name)!.formatterSpec.persistenceUnit.name;
```

Comment:
Not sure you should use `!` here, since there's no guarantee you're going to find an entry there. For example, if the event is fired with `["x", "x"]`, and `getFormat("x")` returns no format props, the first iteration will remove `x` from `_formatSpecsRegistry` and the second one will throw an error due to use of `undefined`

Reply:
I've refactored the code to handle this. .forEach also doesn't work with async functions either so had to use for .. of